python/cudf/cudf/_lib/rolling.pyx

```diff
@@ -75,8 +75,6 @@ def rolling(Column source_column, Column pre_column_window,
                     agg)
             )
     else:
-        if op == "count":
-            min_periods = 0
```

Comment:
There's a few strange things going on here. The rabbit hole starts with the following change in behavior that started in pandas 1.0+:

https://github.com/pandas-dev/pandas/issues/31302

It seems as though in pandas 0.25.3, the `min_periods` parameter was being ignored for `count` aggregations. The justification for this seems to have been that the designers wanted `rolling.count` to behave similarly to other `count` methods elsewhere in the API - e.g. always counting all the non-null values in the group of data in question. It looks like the code here was meant to make `cuDF` do the same. Concretely, this is the previous behavior in pandas:

```
>>> pd.__version__
'0.25.3'
>>> pd.Series([1,1,1,None]).rolling(2, min_periods=2, center=True).count()
0    1.0
1    2.0
2    2.0
3    1.0
dtype: float64
```
Here the `min_periods` parameter ends up getting ignored, evidently in two cases. Firstly, the very first window technically includes the nonexistent value at index position `-1`. Since `min_periods` is meant to be `2`, so we should get a `NaN` here. Secondly, the last value in the original series is `NaN`, and since `center=True` the last value in the resulting series should also be `NaN` as there's no way the window of length `2` can have `2` valid values at that point.

Great I think, so the issue should be solved with Pandas 1.0 since a PR was merged for this, and I can remove this code in cuDF. In doing so we get the answer we'd expect in this case:

```
>>> cudf.Series.from_pandas(pd.Series([1,1,1,None])).rolling(2, min_periods=2, center=True).count()
0    null
1       2
2       2
3    null
dtype: int32
```

But here's what I actually get in Pandas:

```
>>> pd.__version__
'1.0.3'
>>> pd.Series([1,1,1,None]).rolling(2, min_periods=2, center=True).count()
0    NaN
1    2.0
2    2.0
3    1.0
dtype: float64
```
I am not sure if I'm not interpreting what this is supposed to be doing correctly, or what is going on really, I find this quite confusing and am still trying to make sense of it. Perhaps this is still a bug in pandas, but as far as our tests go, we're kind of 'darned if we do, darned if we don't' with these two lines of code.

Reply:
Looking 👍

Reply:
raised https://github.com/pandas-dev/pandas/issues/34466

Reply:
So I at least understand the results we're seeing in this repro around why for instance, the last element is considered a valid entry in this case. My understanding is that in pandas 1.0, it was decided that `count` should respect `min_periods` as it relates to the actual window size - meaning that if an element literally does not exist (is 'off the edge' of the series) then it won't count. For example if the window size is 2, the first element will be NaN because we won't have two elements to compare. 

This is distinct however from the behavior around NaNs for `count`. In pandas, `NaN` doesn't contribute to the `count` of anything:

```
>>> pd.Series([None, 1, 2]).count()
2
```

The pandas devs make the point that this normalizes `rolling.count` against other kinds of `count`s across the codebase. Just to be concrete about this, this particular function breaks from the other rolling functions. For instance `mean` still requires it gets `min_periods` non-nan values. 

```
>>> pd.Series([1,1,1,None]).rolling(2, min_periods=2, center=True).mean()
0    NaN
1    1.0
2    1.0
3    NaN
dtype: float64
>>> pd.Series([1,1,1,None]).rolling(2, min_periods=1, center=True).mean()
0    1.0
1    1.0
2    1.0
3    1.0